File: app/src/main/java/com/senk/bus/data/entity/Route.java
package com.senk.bus.data.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "routes")
public class Route {
    @PrimaryKey(autoGenerate = true)
    public int id;

    @NonNull
    public String name;

    @NonNull
    public String origin;

    @NonNull
    public String destination;

    public boolean isFavorite;

    public boolean isDefault;
}

File: app/src/main/java/com/senk/bus/data/entity/Route.kt
package com.senk.bus.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "routes")
class Route {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    var name: String = ""

    var origin: String = ""

    var destination: String = ""

    var isFavorite: Boolean = false

    var isDefault: Boolean = false
}
